lib/cdo/crowdin/legacy_utils.rb

```diff
@@ -48,19 +48,27 @@ def fetch_changes
       languages = @project.languages
       num_languages = languages.length
       languages.each_with_index do |language, i|
-        language_code = language["code"]
+        language_code = language["id"]
         @logger.debug("#{language['name']} (#{language_code}): #{i}/#{num_languages}")
         @logger.info("~#{(i * 100 / num_languages).round(-1)}% complete (#{i}/#{num_languages})") if i > 0 && i % (num_languages / 5) == 0
 
         etags[language_code] ||= {}
         files = @project.list_files
 
         changed_files = Parallel.map(files, in_threads: MAX_THREADS) do |file|
-          etag = etags[language_code].fetch(file, nil)
-          response = @project.export_file(file, language_code, etag: etag, only_head: true)
+          file_id = file['id']
+          filepath = file['path']
+          etag = etags[language_code].fetch(filepath, nil)
+          response = @project.export_file(file_id, language_code, etag: etag)
           case response.code
           when 200
-            [file, response.headers["etag"]]
+            [
+              filepath,
+              {
+                etag: response["data"]["etag"],
+                download_url: response["data"]["url"]
```

Comment:
The `download_url` returned in the v2 API can only be used once and then it is marked as expired. I think theoretically this should never be an issue because the building of these urls during `fetch_changes` directly precedes the downloading of them.

Of course things find a way to not go as planned, so my thought is to error out the `sync-down` if we receive an expired response during `fetch_changes`. If we don't error out, we will download an error response as a translation file. If this is actually somehow occurring we can implement some type of retry logic or refactor where necessary.

Let me know any thoughts or concerns on this